aim/sdk/objects/plugins/deeplake_dataset.py

```diff
@@ -0,0 +1,113 @@
+
+    def __init__(self, dataset: deeplake.Dataset, auto_commit: bool = True, auto_save_view: bool = True):
+        super().__init__()
+        if not isinstance(dataset, deeplake.Dataset):
+            raise TypeError("dataset must be of type ``deeplake.Dataset``")
+        if dataset.is_view and dataset.has_head_changes:
+            # there is nothing to trace back data to this run.
+            raise ValueError("dataset is a view on a head of uncommitted changes. "
+                             "Consider committing dataset changes before creating a view and logging runs "
+                             "to enable traceability.")
+        if dataset.has_head_changes:
+            if auto_commit:
+                commit_id = dataset.commit(message="autocommit on aim run")
+                logger.info(f'autocommit on run: dataset {dataset.path} with commit id {commit_id}.')
+            else:
+                warnings.warn(
+                    f"Deeplake Dataset {dataset.path} has uncommitted head changes. "
+                    "Consider committing dataset changes before logging runs to enable full traceability.",
+                    UncommittedDatasetWarning,
+                    stacklevel=2,
+                )
+        if dataset.is_view:
+            self.view_info = dataset._get_view_info()
+            view_id = self.view_info.get('id', None)
+            if auto_save_view:
+                try:
+                    vds_path = dataset.save_view(message="auto_save_view on aim run.", id=view_id, optimize=False)
+                    logger.info(f'autosave view on run: dataset {dataset.path} with id {view_id} saved to {vds_path}.')
+                except (NotImplementedError, ReadOnlyModeError) as e:
+                    # views of in-memory datasets and read-only datasets cannot be saved. but keep the view id.
+                    logger.info(f'{str(e)} {dataset.path} with commit id {commit_id}.')
+                    pass
+
+        self.storage['dataset'] = {
+            'source': 'deeplake',
+            'meta': self._get_ds_meta(dataset)
+        }
+
+    def _get_ds_meta(self, ds: deeplake.Dataset):
+        return {
+            "path": ds.path,
+            "commit_id": ds.commit_id,
+            "branch": ds.branch,
+            "has_head_changes": ds.has_head_changes,
+            "pending_commit_id": ds.pending_commit_id if ds.has_head_changes else None,
+            "info": dict(ds.info),  # Info might contain keys such as "description" and "title"
+            "num_samples": ds.num_samples,
+            "max_len": ds.max_len,
+            "min_len": ds.min_len,
+            "is_view": ds.is_view,
+            "view_info": self.view_info if ds.is_view else None,
+            "tensors": {group: self._tensor_meta(tensor) for group, tensor in ds.tensors.items()},
+            "size_approx": ds.size_approx(),
+            "deeplake_version": ds.meta.version
+        }
+
+    def _tensor_meta(self, tensor: deeplake.Tensor):
```

Comment:
A minor comment: there's some amount of code duplication, which should be OK if the `aim.HubDataset` class is going to be removed (see the next comment). Otherwise a small refactoring will be required here.